docs/user-guide/container-environment.md

```diff
@@ -60,7 +60,9 @@ This hook is called immediately before a container is terminated. No parameters
 
 ### Hook Handler Execution
 
-When a management hook occurs, the management system calls into any registered hook handlers in the container for that hook.  These hook handler calls are synchronous in the context of the pod containing the container. Typically we expect that users will make their hook handlers as lightweight as possible, but there are cases where long running commands make sense (e.g. saving state prior to container stop).
+When a management hook occurs, the management system calls into any registered hook handlers in the container for that hook.  These hook handler calls are synchronous in the context of the pod containing the container. This means that for a `PostStart` hook, the container entrypoint and hook will fire asynchronously. However, if the hook takes a while to run or hangs, the container will never reach a "running" state. If a hook fails, it will kill the container.
```

Comment:
This gives a lot of focus to the PostStart hook in a paragraph about all hook handlers. The same note is true for PreStop. I believe if the hook hangs the phase will remain Running and never reach Failed.